ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -123,17 +123,32 @@ public void setBaseSearchDn(final String dn) {
     }
 
     public void setBaseGroupDn(@Nullable final Dn dn) {
-        baseGroupDn = dn != null ? dn : Dn.ROOT;
+        baseGroupDn = dn;
     }
 
     public void setBaseGroupDnString(@Nullable final String dn) {
-        setBaseGroupDn(DnUtils.toDn(dn));
+        setBaseGroupDn(dn != null ? DnUtils.toDn(dn) : null);
     }
 
     public void setMaxPageSize(final int size) {
         this.maxPageSize = size;
     }
 
+    private void assertValidBaseGroupDn() {
+        if (baseGroupDn == null) {
+            throw new UnsupportedOperationException(
+                    "a baseGroupDn needs to be configured before group functionality can be used");
+        }
+    }
+
+    private void assertValidGroupDn(@Nonnull final Dn dn) {
+        assertValidBaseGroupDn();
+        assert baseGroupDn != null;
```

Comment:
no functional/execution value at all, and `assert`s are sometimes not even executed in a JVM.